web/api/v1/api.go

```diff
@@ -703,6 +708,11 @@ func (api *API) labelNames(r *http.Request) apiFuncResult {
 	if names == nil {
 		names = []string{}
 	}
+
+	if len(names) >= limit {
```

Comment:
shouldn't this check be `len(names) == limit` instead of `>=`? In this case the returned elements can be equal to the limit and also contain all elements in the database, but the response will still contain a warning. The same is used on the other two endpoints

Reply:
you mean `len(names) > limit`?
I opened https://github.com/prometheus/prometheus/pull/14116 to fix that.
Thanks and don't hesitate to open an issue for such "issues":)